tests/h2-support/src/trace.rs

```diff
@@ -1,41 +1,19 @@
-use std::{io, str};
 pub use tracing;
 pub use tracing_subscriber;
 
-pub fn init() -> tracing::dispatcher::DefaultGuard {
-    tracing::subscriber::set_default(
-        tracing_subscriber::fmt()
-            .with_max_level(tracing::Level::TRACE)
-            .with_span_events(tracing_subscriber::fmt::format::FmtSpan::CLOSE)
-            .with_writer(PrintlnWriter { _p: () })
-            .finish(),
-    )
-}
-
-struct PrintlnWriter {
-    _p: (),
-}
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
 
-impl tracing_subscriber::fmt::MakeWriter for PrintlnWriter {
-    type Writer = PrintlnWriter;
-    fn make_writer(&self) -> Self::Writer {
-        PrintlnWriter { _p: () }
-    }
-}
-
-impl io::Write for PrintlnWriter {
-    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        let s = str::from_utf8(buf).map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, e))?;
-        println!("{}", s);
-        Ok(s.len())
-    }
-
-    fn write_fmt(&mut self, fmt: std::fmt::Arguments<'_>) -> io::Result<()> {
-        println!("{}", fmt);
-        Ok(())
-    }
+pub fn init() -> tracing::dispatcher::DefaultGuard {
+    let use_colors = atty::is(atty::Stream::Stderr);
+    let layer = tracing_tree::HierarchicalLayer::default()
+        .with_writer(std::io::stderr)
+        .with_indent_lines(true)
+        .with_ansi(use_colors)
+        .with_targets(true)
+        .with_indent_amount(2);
```

Comment:
take it or leave it: perhaps we want to add
```suggestion
        .with_indent_amount(2)
        .with_thread_ids(true)
        .with_thread_names(true);
```
in particular, adding thread names will give us a nice way to know which test the events were logged from, since the thread's name defaults to the name of the test.

Reply:
I added it but I'm not sure this is useful: I only ever look at logs running a single test, not many of them.

Reply:
<img width="1507" alt="Capture d’écran 2022-01-20 à 10 45 20" src="https://user-images.githubusercontent.com/123095/150314017-7f932f7e-0ab2-4e4a-8725-33b563cdcc8d.png">

It's a bit verbose as you can see.

Reply:
hmm, we could remove it --- i suppose that since libtest is capturing the logs now, they'll also be printed out per-test. the only reason to consider having thread names here is if we have tests that spawn multiple threads and they give names to those threads...and we only have one test that spawns threads (https://github.com/hyperium/h2/search?q=thread%3A%3Aspawn).

i'm fine with removing the thread-name configuration, then!